src/nodes/CommandConnections.ts

```diff
@@ -71,10 +113,18 @@ class CommandAdd extends CommandPolykey {
             }),
           );
         } else {
+          // Deep copy the connections array to avoid modifying the original object
+          const sanitizedConnections = JSON.parse(JSON.stringify(connections));
+          // Loop through each connection and remove the sensitive 'authorization' field
+          for (const connection of sanitizedConnections) {
+            if (connection.metadata && connection.metadata.authorization) {
+              delete connection.metadata;
+            }
+          }
```

Comment:
This is a bandaid, not a fix. Find out why metadata is included in the first place. That's the real problem.

Reply:
> This is a bandaid, not a fix. Find out why metadata is included in the first place. That's the real problem.

@tegefaulkes I think the core issue is again of rpc, which is this

    return {
      readable: outputMessageTransformStream.readable,
      writable: inputMessageTransformStream.writable,
      cancel: rpcStream.cancel,
      meta: metadata,
    };
    
The client returns the metadata input. 

How do I go about this?

Reply:
Let's reason through this. We're getting metadata in the output and we don't know why. Where is it coming from? It is a server stream call, so we're getting a stream of the output. 

OK, so in the RPC the first message of the response, or any message can include metadata. Ok so we *do* need to strip out the metadata when reading the output.

But how do we do it? doing a `JSON.parse(JSON.stringify())` is a code smell and should almost never be done since it's a very expensive operation done twice. There are better ways to deep copy.

Finally, why not just strip out the metadata when we're collecting the stream on line 50? You don't need a conditional, you just construct a new object with the information you want.

Reply:
Resolved

Reply:
how was this resolved @addievo ?